Assets/_BForBoss/_Character/Scripts/FirstPersonPlayer.cs

```diff
@@ -46,16 +53,54 @@ protected override void AnimateEye()
             // Base class animates the camera for crouching here, cinemachine handles that
         }
 
+        public override float GetBrakingDeceleration()
+        {
+            if (_slideBehaviour != null && _slideBehaviour.IsSliding)
+            {
+                return _slideBehaviour.brakingDecelerationSliding;
+            }
+            
+            return base.GetBrakingDeceleration();
+        }
+
+        public override float GetMaxSpeed()
+        {
+            if (_slideBehaviour != null && _slideBehaviour.IsSliding)
+            {
+                return _slideBehaviour.MaxWalkSpeedSliding;
+            }
+
+            return base.GetMaxSpeed();
+        }
+
+        protected override Vector3 CalcDesiredVelocity()
+        {
+            if (_slideBehaviour != null && _slideBehaviour.IsSliding)
```

Comment:
Up to you but it could be cleaner to just make a property for this bool (since it being used a lot)

i.e. `private bool IsSliding => _slidebehaviour != null && _slidebehaviour.IsSliding`.

Similarly you could make a method to check

Reply:
then you could position all these methods with ternary operators. i.e. `returning IsSliding ? Vector3.zero " Base.CalcDesiredVelocity()`

Reply:
Made the. changes